webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/PreferencesUtil.java

```diff
@@ -114,6 +125,14 @@ public static void loadPreferences(String aUsername, SettingsService aSettingsSe
                         .getProject());
                 aBModel.setAnnotationLayers(layers);
             }
+
+            // Get color preferences for each layer, init with legacy if not found
```

Comment:
As I said in https://github.com/webanno/webanno/issues/531#issuecomment-310433982 and https://github.com/webanno/webanno/issues/531#issuecomment-310436267 : it might not be the worst idea to change way that the preferences are being saved from the present "manual" way to using Jackson along with changing the datamodel. In such a case, I would probably go with `Map<Long, LayerPreferences>`.